.github/workflows/build-test-lint.yml

```diff
@@ -0,0 +1,87 @@
+name: "Build, Test, Lint"
+on: [push, pull_request]
+jobs:
+  build:
+    strategy:
+      matrix:
+        go-version: [1.22.x]
+        goarch: [amd64]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Set up Go matrix
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v4
+      - name: Build
+        env:
+          GOARCH: ${{ matrix.goarch }}
+          GOOS: ${{ matrix.goos }}
+        run: make build
+# TODO enable when all lint issues are fixed in a separate PR
+#  lint:
+#    runs-on: ubuntu-latest
+#    needs: build
+#    steps:
+#      - name: Set up Go
+#        uses: actions/setup-go@v5
+#        with:
+#          go-version: 1.22.x
+#      - name: Check out code into the Go module directory
+#        uses: actions/checkout@v4
+#      - name: Lint
+#        run: make lint
+  test:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.22.x
```

Comment:
lint & test jobs use static version of go (compared to variable in build matrix). probably minor, but keep it in mind...